direct/tpm2/read_public_test.go

```diff
@@ -51,26 +57,41 @@ func TestReadPublicKey(t *testing.T) {
 		},
 	}
 
+	// Executing the command uses reflection to pack the bytes into a 
+	// TPM2_CreatePrimary command, returns a TPM2_CreatePrimary Response.
+	// This response is also decoded so you are again working with structs 
+	// that can be found in Part 3, Commands, section 24.1.
 	rspCP, err := createPrimary.Execute(thetpm)
 	if err != nil {
 		t.Fatalf("CreatePrimary failed: %v", err)
 	}
 
+	// The TPM can only hold so much in nonvolitile memory, thus we must
+	// flush the handle after we are done using it to prevent overloading. 
+	// Again we defer the flush to after we are done using the object.
 	flushContext := FlushContext{FlushHandle: rspCP.ObjectHandle}
 	defer flushContext.Execute(thetpm)
 
+	// Fill in the ReadPublic struct.
+	// See definition in Part 3, Commands, section 12.4.
 	readPublic := ReadPublic{
 		ObjectHandle: rspCP.ObjectHandle,
 	}
 
+	// Executing the command uses reflection to pack the bytes into a 
+	// TPM2_ReadPublic command, returns a TPM2_ReadPublic Response.
+	// This response is also decoded so you are again working with structs 
+	// that can be found in Part 3, Commands, section 12.4.
 	rspRP, err := readPublic.Execute(thetpm)
 	if err != nil {
 		t.Fatalf("ReadPublic failed: %v", err)
 	}
 
+	// Compare the Unique portion of the PublicAreas to ensure they are equal.
```

Comment:
Might be useful to explain that the Unique represents the public part of the key.

Reply:
Resolved.